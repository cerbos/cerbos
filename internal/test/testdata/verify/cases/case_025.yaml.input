-- testdata/principals.yaml --
---
principals:
  john:
    id: john
    policyVersion: 20210210
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr:
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
tests:
  - name: John and his leave request
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - view:public
        - approve
        - defer
      auxData: myJWT
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW
          approve: EFFECT_DENY
          defer: EFFECT_ALLOW
