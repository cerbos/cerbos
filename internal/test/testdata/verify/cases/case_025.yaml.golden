{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "failed to load test fixtures from testdata: failed to load principals: file is not valid: { /principals/john/policyVersion: [expected string, but got number] }",
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED"
  }
}
